src/areaCalculator/AreaCalculator.java

```diff
@@ -1,11 +1,50 @@
 package areaCalculator;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class AreaCalculator {
 
-    public double calculateAreaOfRectangle(Rectangle rectangle) {
-        double area = rectangle.getLength() * rectangle.getWidth();
-        System.out.println("Area of rectangle (" + rectangle.getLength() + " x "
-                + rectangle.getWidth() + ") = " + area);
-        return area;
+    private List<Shape> shapeList;
+
+    public AreaCalculator() {
+        shapeList = new ArrayList<>();
+    }
+
+    void addShapeToList(Shape shape) {
+        shapeList.add(shape);
+    }
+
+    public void calculateArea(List<Shape> list) {
+        for (Shape shape : list) {
+            String typeOfShape = shape.getClass().getSimpleName();
```

Comment:
More correct way of doing what you  do here is to use `instance of` operator. If you don't know about it you can read more here: https://www.javatpoint.com/downcasting-with-instanceof-operator

Reply:
Applied `instance of` operator